db/redis.js

```diff
@@ -1,123 +1,117 @@
-    var doorStream = rows.map(entry =>  new doorData(entry))
-    return doorStream
+async function getDoorWindow(locationID, startTime, endTime, windowLength) {
+  const rows = await client.xrevrange(`door:${locationID}`, startTime, endTime, 'count', windowLength)
+  const doorStream = rows.map(entry => new DoorData(entry))
+  return doorStream
 }
 
-async function getStatesWindow(locationID, startTime, endTime, windowLength){
-    let rows = await client.xrevrange('state:'+ locationID,startTime, endTime, 'count', windowLength);
-    var stateStream = rows.map(entry =>  new stateData(entry))
-    return stateStream
+async function getStatesWindow(locationID, startTime, endTime, windowLength) {
+  const rows = await client.xrevrange(`state:${locationID}`, startTime, endTime, 'count', windowLength)
+  const stateStream = rows.map(entry => new StateData(entry))
+  return stateStream
 }
 
-async function clearKeys(){
-    await client.flushall()
+async function clearKeys() {
+  await client.flushall()
 }
 
-async function disconnect(){
-    await client.disconnect()
+async function disconnect() {
+  await client.disconnect()
 }
 // POST new door Test data
-const addDoorSensorData = (locationid, signal) => {
-    client.xadd("door:" + locationid,  "*", "signal", signal);
+function addDoorSensorData(locationid, signal) {
+  client.xadd(`door:${locationid}`, '*', 'signal', signal)
 }
 
-async function addDoorTestSensorData(locationid, signal){
-    await client.xadd("door:" + locationid,  "*", "signal", signal);
+async function addDoorTestSensorData(locationid, signal) {
+  await client.xadd(`door:${locationid}`, '*', 'signal', signal)
 }
 
-async function addIM21DoorSensorData(locationid, doorSignal){
-    if(doorSignal==SESSIONSTATE_DOOR.CLOSED||doorSignal==SESSIONSTATE_DOOR.OPEN){
-        await client.xadd("door:"+locationid,  "*","signal", doorSignal)
-    }
+async function addIM21DoorSensorData(locationid, doorSignal) {
+  if (doorSignal === SESSIONSTATE_DOOR.CLOSED || doorSignal === SESSIONSTATE_DOOR.OPEN) {
+    await client.xadd(`door:${locationid}`, '*', 'signal', doorSignal)
+  }
 }
 
-async function addVitals(locationid, signalStrength, cloudDisconnects){
-    client.xadd("vitals:"+locationid, "*", "strength", signalStrength, "cloudDisc", cloudDisconnects)
+async function addVitals(locationid, signalStrength, cloudDisconnects) {
+  client.xadd(`vitals:${locationid}`, '*', 'strength', signalStrength, 'cloudDisc', cloudDisconnects)
 }
 
-const addXeThruSensorData = (locationid, state, rpm, distance, mov_f, mov_s) => {
-    client.xadd("xethru:" + locationid,  "*", 
-        "state", state,
-        "distance", distance,
-        "rpm", rpm, 
-        "mov_f", mov_f,
-        "mov_s", mov_s, 
-    );
+function addXeThruSensorData(locationid, state, rpm, distance, mov_f, mov_s) {
+  client.xadd(`xethru:${locationid}`, '*', 'state', state, 'distance', distance, 'rpm', rpm, 'mov_f', mov_f, 'mov_s', mov_s)
```

Comment:
This actually apparently comes from eslint's `function-call-argument-newline` as well as prettier so I think I would have to disable that for the entire file (which I personally think is ok since it's happy with everything else as is), each individual line of that block in-line, orrrr probably the best option which is still not very pretty is below 😬 

```
// prettier-ignore
/* eslint-disable function-call-argument-newline */
function addXeThruSensorData(locationid, state, rpm, distance, mov_f, mov_s) {
  client.xadd(
    `xethru:${locationid}`, '*', 
    'state', state, 
    'distance', distance, 
    'rpm', rpm, 
    'mov_f', mov_f, 
    'mov_s', mov_s)
}
/* eslint-enable function-call-argument-newline */
```

I tried to see if that disable/enable can be after `//` instead as it looks slightly nicer but alas, no

Reply:
Sure, I think that looks OK. It would be nice to have a comment explaining why all the ignores.